src/resolvers/projectResolver.ts

```diff
@@ -333,21 +333,38 @@ export class ProjectResolver {
     searchTerm?: string,
   ) {
     if (!searchTerm) return query;
+    const SIMILARITY_THRESHOLD = 0.4;
 
     return query.andWhere(
       new Brackets(qb => {
-        qb.where('project.title ILIKE :searchTerm', {
-          searchTerm: `%${searchTerm}%`,
-        })
-          .orWhere('project.description ILIKE :searchTerm', {
-            searchTerm: `%${searchTerm}%`,
-          })
-          .orWhere('project.impactLocation ILIKE :searchTerm', {
+        qb.where(
+          'SIMILARITY(project.title, :searchTerm) > :similarityThreshold',
```

Comment:
To do efficient searching, it's necessary to index appropriately index target fields. Hopefully, these target fields are not updated frequently, and indexing them won't put a substantial load on the DB.

Reply:
Given this explanation:
>- If you're looking for approximate matches between strings and don't necessarily care about word boundaries, similarity might be sufficient.
> - If you want to consider individual words within the strings and prefer a more granular similarity calculation, word_similarity could be a better option.
> - If you want to ensure that matches occur within complete word boundaries and avoid partial word matches, strict_word_similarity might be more appropriate.

It seems work_similarity function would be a better fit for our use case, @sembrestels wdyt?